lib/components/LoggedInAs.js

```diff
@@ -67,18 +68,33 @@ var LoggedInAs = React.createClass({
   },
 
   renderChooseDevices: function() {
+    var title = '';
     var uploadInProgress = this.props.isUploadInProgress;
+    var isDisabled = uploadInProgress;
 
-    if(this.props.isClinicAccount){
+    if (this.props.isClinicAccount) {
       return null;
     }
+
+    if (_.isEmpty(this.props.targetUsersForUpload) ||
+        this.props.targetUsersForUpload.length < 1) {
```

Comment:
Heh, yeah, there might have been a stage when we weren't defaulting `targetUsersForUpload` to an empty array I suppose? Go ahead and change there too.